src/studio/src/designer/backend/Services/Implementation/RepositorySI.cs

```diff
@@ -502,49 +510,47 @@ public List<string> GetLanguages(string org, string app)
         /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
         /// <param name="app">Application identifier which is unique within an organisation.</param>
         /// <param name="texts">The texts to be saved</param>
-        public void SaveServiceTexts(string org, string app, Dictionary<string, Dictionary<string, string>> texts)
+        public void SaveServiceTexts(string org, string app, Dictionary<string, Dictionary<string, TextResourceElement>> texts)
         {
-            // Language, key, value
-            Dictionary<string, Dictionary<string, JObject>> resourceTextsAsJson =
-                new Dictionary<string, Dictionary<string, JObject>>();
+            // Language, key, TextResourceElement
+            Dictionary<string, Dictionary<string, TextResourceElement>> resourceTexts =
```

Comment:
What is the difference between this dictionary and the input parameter?

Reply:
Input is on the form: <textResourceElement.Id <language, textResourceElement>>
The new dictionary is on the form <Language, <<textResourceElement.Id, TextResourceElement>>

Reply:
I see